src/snmalloc/backend_helpers/smallbuddyrange.h

```diff
@@ -141,6 +141,22 @@ namespace snmalloc
       UNUSED(k, size);
       return true;
     }
+
+    /**
+     * Erases the pointers stored in this node when we are removing it from
+     * the RBTree. Useful so that returned memory does not leak internal
+     * pointers.
+     * @ptr must not be NULL and must not point to a NULL node.
+     */
+    static void clear(Contents p)
+    {
+#ifdef SNMALLOC_CLEAN_POINTERS
```

Comment:
I am a bit uncomfortable about adding new `#ifdef`s, since we're trying to remove the existing ones - they can introduce ODR violations if we have multiple uses of snmalloc in a single program.  Ideally this would be a constexpr property of the back end.  If that isn't feasible then at least making them constexpr flags in `allocconfig.h` would be better.  

I think the ideal approach would probably be:

 - Use the macro to set a `constexpr` flag in `allocconfig.h`.
 - Make the back end Options structure have a flag that defaults to whatever the value in allocconfig.h is.
 - Use `if constexpr` with the back-end flag everywhere in the front end.

Reply:
I don't see that having `ifdef`s that define `static constexpr`s is going to help with ODR violations, but I think putting the `ifdef`s in as few places as possible, and using `if constexpr` makes the code easier to read.